wolfcrypt/src/misc.c

```diff
@@ -39,11 +39,18 @@
  */
 
 #ifdef NO_INLINE
+    #undef  WOLFSSL_MISC_COMPILED
+    #define WOLFSSL_MISC_COMPILED
     #define STATIC
 #else
     #define STATIC static
 #endif
 
+/* Check for if compiling misc.c when not needed. FIPS build has source code
+   without preprocessor flag so the warning is not used in a FIPS build. */
+#if !defined(WOLFSSL_MISC_COMPILED) && !defined(HAVE_FIPS)
```

Comment:
I think we'll no longer need to worry about FIPS if above guidance is followed.  The check more simply becomes 

! WOLFSSL_MISCC_INCLUDED && ! NO_INLINE